### ISSUE TITLE
CheCluster CR should not be created in openshift-operators namespace

### ISSUE THREAD

**Issue by l0rd** [eclipse/che#21399]
### Is your enhancement related to a problem? Please describe

Che operands are deployed in `openshift-operators` namespace. That namespace, as it's name suggests, is dedicated to operators and not to operands. We should not deploy postgres, database, registries, server, dashboard etc...there.

### Describe the solution you'd like

`CheCluster` CR should be created in `eclipse-che` namespace as soon as the operator is deployed. 

That also means that an admin doesn't have to manually create a `CheCluster` after Che operator subcription: it's automatically created (1 step installation instead of 2 steps installation).

### Additional information

OpenShift GitOps operator does that: it creates the `openshift-gitops` namespace if it doesn't exist and 
Red Hat product OpenShift Dev Spaces default `CheCluster` namespace should be `openshift-devspaces`.

**Comment by ibuziuk** [comment]
> Red Hat product OpenShift Dev Spaces default CheCluster namespace should be openshift-devspaces

probably we should reconsider the `openshift-` prefix since it will be impossible to use such namespaces on OSD -  (^kube-.*|^openshift.*|^default$|^redhat-.*)"

**Comment by l0rd** [comment]
> > Red Hat product OpenShift Dev Spaces default CheCluster namespace should be openshift-devspaces
> 
> probably we should reconsider the `openshift-` prefix since it will be impossible to use such namespaces on OSD - (^kube-._|^openshift._|^default$|^redhat-.*)"

Yes, I would do what other openshift prefixed products do with their operands (pipelines, service mesh, serverless, gitops, sbo etc...)

**Comment by l0rd** [comment]
We have checked with @ibuziuk and, when deployed on OSD, OpenShift GitOps creates a `openshift-` prefixed namespace. So that's what Che/DevSpaces should do too.

**Comment by l0rd** [comment]
[Downstream issue](https://issues.redhat.com/browse/CRW-2997)

**Comment by tolusha** [comment]
@l0rd 
I have some concerns about CheCluster creation by the operator. I do believe it will lead having 2 CR (the first one created by the operator, another one - created by user via OpenShift UI)

So, I'm thinking of a slightly different approach. Let's keep the current behavior but, If user creates `CheCluster` CR in `openshift-operators` namespace, then operator moves it to a right place (creates namespace if needed). 

WDYT ?

**Comment by l0rd** [comment]
This is a good point @tolusha. 

I would not move a CR created by a user. It would be hard to follow what's going on. 

This is how it works for OpenShift Gitops:

1. Click install in OperatorHub
<img width="600" alt="image" src="https://user-images.githubusercontent.com/606959/172647597-59301600-2810-46d8-b495-745965d86fc8.png">
2. Select the "Channel" and the "Update approval"
<img width="600" alt="image" src="https://user-images.githubusercontent.com/606959/172647898-71e61dd0-fa35-4dff-976e-c977b7614755.png">
3. A completion message appears
<img width="600" alt="image" src="https://user-images.githubusercontent.com/606959/172648369-8d6003e5-1cd3-4f25-80a7-82f3bd9ab9e7.png">
4. An ArgoCD instance has been created automatically in namespace `openshift-gitops`
<img width="600" alt="image" src="https://user-images.githubusercontent.com/606959/172654073-c557165d-c3ae-4225-8764-b8631e67c3fb.png">
5. Users can still click on "View Operator" and create a new "ArgoCD" instance
<table>
<tr>
	<td><img width="300" alt="image" src="https://user-images.githubusercontent.com/606959/172649146-761c2111-0b76-4903-95e2-ed95bd18ae9f.png">
	<td><img width="300" alt="image" src="https://user-images.githubusercontent.com/606959/172649824-39749e6d-0110-4126-9bfe-d94c027da29c.png">
</table>

For me that's easier than what we currently have. But I will review it with @RickJWagner and @Kasturi1820 .

Two ways to make it even better are:
1. Print Che URL in the completion message at step 3
2. Provide an error message if a user tries to create a second CheCluster in the OpenShift cluster similar to this one:
<img width="848" alt="image" src="https://user-images.githubusercontent.com/606959/172655969-aea4616a-1a20-43ec-83ad-6aa2935d923e.png">
